src/yang/ietf-bgp-policy.yang

```diff
@@ -272,6 +300,95 @@ module ietf-bgp-policy {
         }
       }
     }
+
+    container prefix-sets {
+      description
+        "Definitions for a list of IPv4 or IPv6 prefixes which
+         are matched as part of a policy.";
+
+      list prefix-set {
+        key "name";
+        description
+          "List of the defined prefix sets";
+
+        leaf name {
+          type string;
+          description
+            "Name of the prefix set used to reference the set in
+             match conditions.";
+        }
+
+        leaf mode {
+          type identityref {
+            base prefix-mode;
+          }
+          description
+            "Indicates the mode of the prefix set, in terms of
+             which address families (IPv4, IPv6, or both) are
+             present. The mode provides a hint, but the device
+             must validate that all prefixes are of the indicated
+             type, and is expected to reject the configuration if
+             there is a discrepancy.  The 'mixed' mode may not be
+             supported on devices that require prefix sets to be
+             of only one address family.";
+        }
+      }
+
+      container prefixes {
+        description
+          "Enclosing container for the list of prefixes in a policy
+           prefix list.";
+
+        list prefix {
+          key "ip-prefix masklength-range";
```

Comment:
A question for Keyur is do we think that implementations that support disjoint overlapping prefixes with range have consistent implementation for lookup.